src/domains/skills/skills.service.js

```diff
@@ -0,0 +1,38 @@
+import { skillsResource } from './skills.resource';
+
+class SkillsService {
+  async create(createSkillBody) {
+    const skill = await skillsResource.create(createSkillBody);
+    return skill;
+  }
+
+  async getSkills(reqQuery) {
+    let skills;
+    if (reqQuery.name) {
```

Comment:
I don't know if this would make sense, but we could definitely use it here https://www.codementor.io/uditrastogi/replace-conditional-statements-if-else-or-switch-with-polymorphism-ryl8mx4ns, replace-conditional-statements-with-polymorphism

Reply:
I see the point here. I tried to make the method robust so that we just have one endpoint `GET /skills` that can accept either `name`, `id`, `tags` as query strings or if you don't supply any of them, it returns `the first 10 skills` in the database -- if also `limit` and `offset` is not supplied in the query. it would be a bit challenging to use polymorphism here.

 I am not saying there is no better way of doing it though, am open to more suggestions.